sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/Schema.java

```diff
@@ -208,6 +209,7 @@ public int hashCode() {
     public static final Set<TypeName> STRING_TYPES = ImmutableSet.of(STRING);
     public static final Set<TypeName> DATE_TYPES = ImmutableSet.of(DATETIME);
     public static final Set<TypeName> CONTAINER_TYPES = ImmutableSet.of(ARRAY);
+    public static final Set<TypeName> MAP_TYPES = ImmutableSet.of(MAP);
```

Comment:
<!--thread_id:cc_180705569_t; commit:7bf9d2048e8d568ee9280629db391775cc490358; resolved:0-->
<!--section:context-quote-->
> **XuMingmin** wrote:
> would separate here, CONTAINER should be ARRAY/SET. `List<KV<>>` could be a CONTAINER_TYPE, `Map<>` is not.

<!--section:body-->
I'm still confused about this. In most systems, Map is considered a container (e.g. in Java Map is a container type)

Reply:
In Java, container extends `Collection` and map extends `Map`, they're very different IMO. If we merge them together I don't see any benefit as this is a backend function and developers are using either `TypeName.ARRAY.type().withComponentType()` or `TypeName.MAP.type().withMapType`. 

To make it clear, I would prefer to use the term **Collection** instead of *Component* or *Contianer*. Any comments?